src/ui/control/geolocate_control.js

```diff
@@ -251,9 +263,31 @@ class GeolocateControl extends Evented {
 
     _updateMarker(position: ?Position) {
         if (position) {
-            this._userLocationDotMarker.setLngLat([position.coords.longitude, position.coords.latitude]).addTo(this._map);
+            const center = new LngLat(position.coords.longitude, position.coords.latitude);
+            this._accuracyCircleMarker.setLngLat(center).addTo(this._map);
+            this._userLocationDotMarker.setLngLat(center).addTo(this._map);
+            this._accuracy = position.coords.accuracy;
+            this._onZoom();
```

Comment:
My personal preference would be to wrap all the checks you have in `_onZoom` inside `_updateCircleRadius`, and then just call `_updateCircleRadius` here and also `on('zoom')`.

Since you're calling _onZoom for something that isn't due to a zoom event, I think changing this would make the code slightly more readable.

Reply:
Though I guess that's an opinionated change too, since now you're calling `_updateCircleRadius` even though you might not be updating it (based on the check). I'm not sure what's best, but I just think it's best not to have onZoom triggered by something that isn't a zoom change.

Reply:
Good point, agreed 👍